rc/core/formatter.kak

```diff
@@ -11,7 +17,10 @@ define-command format -docstring "Format the contents of the current buffer" %{
                 evaluate-commands %sh{
                     readonly path_file_out=\$(mktemp \"${TMPDIR:-/tmp}\"/kak-formatter-XXXXXX)
 
-                    if cat \"${path_file_tmp}\" | eval \"${kak_opt_formatcmd}\" > \"\${path_file_out}\"; then
```

Comment:
Not sure about `lintcmd`, but I think `formatcmd` should stay. I can't think of a formatter that can't use stdin/stdout, it's the obvious interface.

Sometimes we need to pass additional information to the formatter, like

```kak
hook global WinSetOption filetype=(c|cpp|java) %{
	set-option window formatcmd "clang-format -style=file -assume-filename=%val{buffile}"
}
```

but that's independent of this change

Reply:
The point of this PR is to avoid having the editor assume that whatever tool the user might pick will support being called in the way prescribed in `.kak` scripts.

Even though your remark is accurate, it also applies to linters (most of them certainly support input on the standard stream), and it's the same assumption (that is already being made by the editor at the moment) that led to my making the PR.

I think it's better to be consistent.